protocol/strict-derivation.md

```diff
@@ -0,0 +1,355 @@
+[already discussed](https://www.notion.so/oplabs/Handling-Invalid-Batches-48bb972368ce409c84bd9189eddd0577)
+and for simplicity it was decided to use the atomic validity model. However, above discussion
+focusses on implications of the atomic validity model for EL-sync. We're now concerned with behavior
+around CL client restarts and reorgs.
+
+## Proposed Design
+
+To resolve open questions regarding the implementation of Partial Span Batch Validity, I propose to
+apply the principle to consistently change the validation rules to be only forward-invalidating, and
+thus treat a span batch like an additional (optional) stage in the derivation pipeline that sits
+before the batch queue, so that the batch queue pulls singular batches from this previous span batch
+stage. This way, we never backwards-invalidate the span batch, only forward.
+
+This means that sequencer drift validations are applied individually to each singular batch derived
+from the span batch. On a failed check, due to Steady Batch Derivation, the singular batch is
+immediately derived as a deposit-only block, and the remaining span batch is invalidated.
+
+There are still checks that naturally apply to the whole span batch due to its data structure:
+- A span batch only has one parent reference, since all internal batch references are implicit.
+Therefore, a failed parent check invalidates the whole span batch.
+- A failed sequencing window check will always apply to the oldest batches in a span batch, so a
+failed check of the first batch will invalidate the full span batch.
+- If `span_end.timestamp < next_timestamp` (as defined in the Span Batch specs), the whole batch is
+dropped, as it doesn't contain any new batches (this would also happen if applying timestamp checks
+to each derived singular batch individually).
+
+Those checks are done once per span batch before the batch queue derives any singular batches.
+
+### Mid-way restart
+
+One concern that was raised with partial span batch validity is how to deal with mid-way restarts,
+i.e. when the CL client restarts, losing its internal derivation pipeline state while the safe head
+of the EL client is in the middle of a span batch. In particular, the correct behavior needs to
+cover the case where a L1 reorg happened in the meantime that reorged out the L1 block that
+contained the span batch.
+
+I fail to see currently how partial span batch validity makes this case worse or different. With
+atomic span batch validity, there's the pending-safe state of derived blocks that might be reorged
+out later by a backwards-invalidating singular batch derived from a span batch. So if we only
+forwards-invalidate, this becomes easier.
+
+And in any case, a restart or reorg needs to reconstruct the derivation pipeline state from the new
+(sync-start) L1 block, so I don't understand how reconstructing span batch derivation state is
+different from reconstructing the channel bank and batch queue in general. If a restart or reorg
```

Comment:
I think you're correct here. The span-batch as a whole should be retrievable, assuming we always revisit the channel-window fully.